user-bag-service/src/main/kotlin/io/baum/userbagservice/users/web/UserController.kt

```diff
@@ -0,0 +1,30 @@
+package io.baum.userbagservice.users.web
+
+import arrow.core.getOrHandle
+import io.baum.userbagservice.users.tools.hideSensitiveFields
+import io.baum.userbagservice.users.web.client.ApiClientFactory
+import io.baum.userbagservice.users.web.model.UserModel
+import org.springframework.http.HttpStatus
+import org.springframework.web.bind.annotation.*
+
+@RestController
+@RequestMapping("/users")
+class UserController(
+    private val userService: UserService,
+    private val apiClient: ApiClientFactory
+) : CreateUserApi, RetrieveUserApi {
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    override fun createUser(@RequestBody user: UserModel) = userService.createUser(user)
+
+    @GetMapping("/{id}")
+    override fun getUserById(@PathVariable id: String, @RequestParam password: String) = userService.getUserById(id, password)
```

Comment:
Get by password sounds like a bit of a security hole... You might want to change that to email or something to avoid scaring your future self when you look back haha

Reply:
Or alternatively have it as part of your request body if this is some sort of login step

Reply:
@ed0906 I knew right from the get-go that it's a security risk to do it like that and my intention was to put something more secure in once the current scope is completed. Considering it's all fake people with fake data, I'm not losing too much sleep right now but will look into this in the future